src/index.ts

```diff
@@ -127,6 +129,15 @@ const mergeOptions = <A, B>(
     params: merge(left.params, right.params),
   })
 
+const fnToPromise = <T>(fn: () => T | Promise<T>): Promise<T> => {
+  try {
+    return Promise.resolve(fn())
+  } catch (e) {
+    // pick up synchronous throws
+    Promise.reject(e)
```

Comment:
I think you should return `reject`, otherwise the error will not be caught. 
This code also can be written in more lightweight way. 

```ts
const fnToPromise = <T>(fn: () => T | Promise<T>): Promise<T> => Promise.resolve().then(fn)
```

Reply:
good catch! 💯🙌